Java/me/dilley/MineStat.java

```diff
@@ -79,7 +79,7 @@
 
   public MineStat(String address, int port)
   {
-    this(address, port, DEFAULT_TIMEOUT);
+    this(address, port, DEFAULT_TIMEOUT * 1000);
```

Comment:
Hm. I'd think that this would introduce another bug:

```java
this(address, port, DEFAULT_TIMEOUT * 1000);
```

This calls the 2nd overloaded constructor, `public MineStat(String address, int port, int timeout)`, which calls `setTimeout(timeout);`. Within `setTimeout()`, the following statement exists which would again multiply the value by `1000`:

```java
this.timeout = timeout * 1000;
```

This would cause `timeout` to ultimately be `5000000` rather than `5000`.

Reply:
In this context, `getTimeout()` would also yield an incorrect value. If this method is called rather than directly accessing the `timeout` value, the value is again multiplied by `1000`. Feel free to modify the code to fix this issue, @grimsi. I propose that we either:
1. Set and store the value properly once (within the constructor) and remove the calculations in `getTimeout()` and `setTimeout()`.
Or:
2. Simply keep any value passed as the whole number (`5` for example), but only handle the math in `getTimeout()` and call this method to get at the timeout value.

Does the above make sense to you?

Reply:
My bad, I just realized that I made a mistake. I modified the class by removing the getters/setters and using Lombok instead. I didn’t see that getTimeout and setTimeout have additional logic in them. Will this makes my bug report unnecessary I would still like to propose that the logic should be moved away from the getter/setter

Reply:
No worries, @grimsi. That works for me. If you want to push an update to your PR, I can get it approved.

Reply:
Just pushed an update which should fix the bug introduced by me.